src/main/java/com/palantir/gradle/versions/AddVirtualPlatformPlugin.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.NamedDomainObjectProvider;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.plugins.JavaPlugin;
+
+public class AddVirtualPlatformPlugin implements Plugin<Project> {
+
+    private static final String ADD_VIRTUAL_PLATFORM_CONSTRAINTS_PROPERTY =
+            "com.palantir.gradle.versions.addVirtualPlatformConstraint";
+    static final String VIRTUAL_PLATFORM_NAME = "palantir-virtual-platform";
```

Comment:
Whatever we choose as our identifier here, we should refer to (gradle-)consistent-version rather than palantir.